lib/saml_idp/logout_request_builder.rb

```diff
@@ -15,6 +15,10 @@ def initialize(response_id, issuer_uri, saml_slo_url, name_id, algorithm)
       self.algorithm = algorithm
     end
 
+    def reference_id
+      self.response_id
```

Comment:
This is overriding the default generate method call that you referenced to use the attribute set by the constructor.

I'll see if I can put this into `LogoutBuilder`.